packages/next/server/config-shared.ts

```diff
@@ -145,6 +146,7 @@ export interface ExperimentalConfig {
    * [webpack/webpack#ModuleNotoundError.js#L13-L42](https://github.com/webpack/webpack/blob/2a0536cf510768111a3a6dceeb14cb79b9f59273/lib/ModuleNotFoundError.js#L13-L42)
    */
   fallbackNodePolyfills?: false
+  upsteamHttpProxyConfig?: HttpProxyOptions
```

Comment:
Note: this should only be exposing the experimental `proxyTimeout` config and I don't think we want to expose all initially so that we have correct defaults.

Reply:
@ijjk I wasn't looking to solve that particular issue with the proxy timeout. Rather, it was to solve
https://github.com/vercel/next.js/discussions/37875 and https://github.com/vercel/next.js/discussions/19078
My use case required both adding a header to the proxy as well as changing other settings on the proxy. 
Given the wide range of scenarios that might be needed to connect to an upstream, IMO it makes more sense to expose the whole config rather than add piecemeal individual settings.

Reply:
Agreed. Only being able to access the timeout config is rather limited, since a main use case seems to be things like custom headers.